### PR TITLE
Add project structure

### PR DESCRIPTION
**Pull request by dkryvi**

### CODE REVIEW

File: lib/middleware/error_handler.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+const errorHandler = (err, req, res, next) => { // eslint-disable-line no-unused-vars
+  const statusCode = getErrorStatusCode(err);
+  const data = pick(err, 'message', 'errors');
+
+  res.status(statusCode).json(data);
+};
+
+errorHandler.notFound = () => {
+  throw new httpErrors.NotFound('Resource not found');
+};
```
Comment (akosenko) [inline lib/middleware/error_handler.js]: unused code

File: lib/routes/api.v1.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import express from 'express';
+
+const router = express.Router();
+
+router.get('/', (req, res) => res.send('Hello world!!!'));
+
+export default router;
```
Comment (akosenko) [inline lib/routes/api.v1.js]: api.v1.js - strange name

**Comment by akosenko** [comment]
I think that we should rename package lib to app

File: lib/server.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import app from './services/app';
+
+const PORT = 3000;
+
+app.listen(PORT, (err) => {
+  if (err) {
+    console.log(err);
+  } else {
+    console.log(`Server listen on port: ${PORT}`);
```
Comment (akosenko) [inline lib/server.js]: what about logger?

File: lib/server.js [hunk 4]
```diff
@@ -0,0 +1,11 @@
+import app from './services/app';
+
+const PORT = 3000;
```
Comment (akosenko) [inline lib/server.js]: pleasse create configuration file and move all such variables to them

File: lib/services/app.js [hunk 5]
```diff
@@ -0,0 +1,12 @@
+import express from 'express';
+
+import errorHandler, { notFound } from '../middleware/error_handler';
+import apiv1 from '../routes/api.v1';
+
+const app = express();
+
+app.use('/api/v1', apiv1);
+app.use('/', notFound);
+app.use(errorHandler);
```
Comment (akosenko) [inline lib/services/app.js]: I'm prefer to use separate routes, for example:
routes.map(require).forEach(function (route) {
  route(app);
});
what do you think?

File: lib/services/app.js [hunk 6]
```diff
@@ -0,0 +1,12 @@
+import express from 'express';
+
+import errorHandler, { notFound } from '../middleware/error_handler';
```
Comment (akosenko) [inline lib/services/app.js]: I'm not sure that it is correct import

File: lib/server.js (hunk 3 above)
Comment (Vanish404) [reply lib/server.js]: I think, this pr is about project structure, but not about logger:)
As for me we need to move a feature about logger in separate pr. Step by step. @inst-event-emitter team, what do you think?

Comment (akosenko) [reply lib/server.js]: ok, I'll create new issue

File: lib/services/app.js (hunk 5 above)
Comment (Vanish404) [reply lib/services/app.js]: @akosenko I'm not sure that I understand you. Would you might writing a example code with that preferences?

File: package.json [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+  "name": "eventemitter-backend",
+  "version": "1.0.0",
+  "description": "Eventemitter server side",
+  "main": "lib/server.js",
+  "repository": "https://github.com/inst-event-emitter/eventemitter-backend.git",
+  "author": "DenKrivichanin <krivichaninds@gmail.com>",
+  "license": "MIT",
+  "scripts": {
+    "start": "nodemon --exec yarn run babel-node -- lib/server.js --presets es2015,stage-2"
+  },
```
Comment (Vanish404) [inline package.json]: You can specify the version of node, like that:
```{ "engines" : { "node" : "8.10.0" } }```

File: lib/services/app.js (hunk 5 above)
Comment (akosenko) [reply lib/services/app.js]: well, lets imagine that you have two routes - event.js and account.js in routes directory. To initialize them we'll write smth like that
app.js
------------------------------------------------------
const routes = globSync('./app/routes/*.js', {
  cwd: __dirname
}).map(require);
routes.forEach(function (route) {
  route(app);
});

account.js
------------------------------------------------------
module.exports = function (app) {
  const accountRouter = express.Router();
  accountRouter.get
  accountRoute.post
  ....
  app.use('api/v1', accountRouter);
}

Comment (akosenko) [reply lib/services/app.js]: this approach will provide modularity and easy way to write tests for each route

File: lib/middleware/error_handler.js (hunk 1 above)
Comment (dkryvi) [reply lib/middleware/error_handler.js]: This part of code emit error which is handling by error_handler middleware

File: lib/routes/api.v1.js (hunk 2 above)
Comment (dkryvi) [reply lib/routes/api.v1.js]: [Resolved](https://github.com/inst-event-emitter/eventemitter-backend/pull/1/commits/980a9aff36919516137b8b09e433e0636a868326#diff-4c20e13ad2ba66d12b169e0a032a9504L8)

File: lib/services/app.js (hunk 6 above)
Comment (dkryvi) [reply lib/services/app.js]: [Resolved](https://github.com/inst-event-emitter/eventemitter-backend/pull/1/commits/29ee8781536f4769deee22e885fe69b0ccf58ca9#diff-4c20e13ad2ba66d12b169e0a032a9504L3)

File: lib/server.js (hunk 4 above)
Comment (dkryvi) [reply lib/server.js]: [Resolved](https://github.com/inst-event-emitter/eventemitter-backend/pull/1/commits/2ebc07740b45973e3a84e5c2f649005a9b97b061)

File: lib/services/app.js (hunk 5 above)
Comment (dkryvi) [reply lib/services/app.js]: It's a nice idea to move routes with the same part of route name to own files

**Review by akosenko — APPROVED** [review]

**Pull request merged**